apps/files/utils.py

```diff
@@ -542,6 +542,10 @@ def parse_xpi(xpi, addon=None, check=True):
 def check_xpi_info(xpi_info, addon=None):
     from addons.models import Addon, BlacklistedGuid
     guid = xpi_info['guid']
+    if amo.get_user():
+        user_id = amo.get_user().id
+    else:
+        user_id = 0
```

Comment:
I'll try another code (because obviously the one above doesn't make sense). Maybe I shouldn't review code when I can't think.
How about
```python
if amo.get_user():
    guid_clashes = Addon.unfiltered.exclude(authors__id=amo.get_user()).filter(guid=guid)
else:
    guid_clashes = Addon.objects.filter(guid=guid)
```

And then use `guid_clashes.exists()` below?